### DISCUSSION TITLE
Assistant chat

### DISCUSSION THREAD

**Discussion by oresteafrica** [openai-php/client#322] [Q&A] [answered]
Thanks a lot for making this PHP library available.
I am trying to use it in order to work with a predefined assistant.
Assistant API chat flow for sending and receiving messages to and from the assistant requires a loop on the call api.openai.com/v1/threads/{threadId}/runs/{runId} until it returns run.status = “completed”.
How do you solve it with you library?
Thanks for the attention.
Kind regards.

**Comment by gehrisandro** [comment, marked as answer]
You can find one way to solve this in my blog article: https://gehri.dev/blog/how-to-use-the-openai-assistants-api-in-php-and-laravel
